### DISCUSSION TITLE
Is it possible to add variable substitution in markdown files?

### DISCUSSION THREAD

**Discussion by miniscruff** [squidfunk/mkdocs-material#6922] [Q&A] [answered]
I am transitioning from hugo to mkdocs and as part of my process I use some reflection to generate docs from my configuration. During this process I also want to have a link to the source code for that config value and that requires me to know the current version of the application. My previous strategy was to use a hugo shortcode which has access to the sites params ( example `{{.Site.Params.version}}` ) that I updated as part of my release process ( in the `config.toml` that hugo uses which is basically the same idea as `mkdocs.yml` )

I tried to replicate this feature by building the same markdown like syntax in the meta object of the mkdocs.yml file but as you would imagine by my post here that does not seem to be substituted during the build.

Here is the relevant part of the config.
```yaml
# mkdocs.yml
meta:
  # I believe version is used by the versioning
  # plugin so I used a different key here
  current_version: v1.2.3
```

Here is what I tried to do in my config markdown file
```md
<!-- page markdown -->
.../changie/blob/{{ config.extra.current_version }}/...
```

End goal, replace the current version with the value from the mkdocs.yml
```md
.../changie/blob/v1.2.3/...
```

The only idea I have left is some how creating a custom template? Its not really a page template though its like for every value sort of thing?

**Comment by squidfunk** [comment, marked as answer]
I think the [macros plugin](https://mkdocs-macros-plugin.readthedocs.io/) is what you're looking for ☺️

**Reply by miniscruff** [reply]
You know... I definitely looked for a while and I knew the whole time I had the wrong terms. Let me whip this up and I can mark it as closed...

**Reply by squidfunk** [reply]
No worries. We don't have an explicit mention for it on our documentation and it can be tough to find resources related to MkDocs sometimes. I recommend to check out the [catalog](https://github.com/mkdocs/catalog) – it's pretty comprehensive.

**Reply by miniscruff** [reply]
Ok awesome, will keep that in mind. For anyone else here coming from a Go background you may want to change the j2 start/end block to not conflict with the Go templating. My config for any future readers.
```yaml
  - macros:
      j2_variable_start_string: "<<"
      j2_variable_end_string: ">>"
```
